### PR TITLE
offline `plot` command

### PR DESCRIPTION
**Pull request by chriddyp**
Description: Write Plotly graphs to an HTML file from Python

### CODE REVIEW

**Comment by chriddyp** [comment]
Here's a first pass at `plotly.offline.plot`
offline-plot
- [ ] tests

hey @cldougl @theengineear - could you check this out?

File: plotly/offline/offline.py [hunk 1]
```diff
@@ -103,76 +170,118 @@ def iplot(figure_or_data, show_link=True, link_text='Export to plot.ly',
...
+    import plotly.graph_objs as go
 
-    # TODO: The get_config 'source of truth' should
-    # really be somewhere other than plotly.plotly
-    plotly_platform_url = plotly.plotly.get_config().get('plotly_domain',
-                                                         'https://plot.ly')
-    if (plotly_platform_url != 'https://plot.ly' and
-            link_text == 'Export to plot.ly'):
+    plot([
+        go.Scatter(x=[1, 2, 3], y=[3, 2 6])
```
Comment (cldougl) [inline plotly/offline/offline.py]: missing comma `y=[3, 2, 6]`

**Comment by cldougl** [comment]
@chriddyp first pass looks good- all the defaults are working well for me!

File: plotly/offline/offline.py [hunk 2]
```diff
@@ -103,76 +170,118 @@ def iplot(figure_or_data, show_link=True, link_text='Export to plot.ly',
...
-    else:
-        width = str(width) + 'px'
 
-    plotdivid = uuid.uuid4()
-    jdata = json.dumps(figure.get('data', []), cls=utils.PlotlyJSONEncoder)
-    jlayout = json.dumps(figure.get('layout', {}), cls=utils.PlotlyJSONEncoder)
+def plot(figure_or_data,
+         show_link=True, link_text='Export to plot.ly',
+         validate=True, output_type='file',
+         include_plotlyjs=True,
```
Comment (chriddyp) [inline plotly/offline/offline.py]: In the future, when [plotly.js is modularized](https://github.com/plotly/plotly.js/issues/41), we might allow folks to include sub-bundles here. Maybe something like `plotlyjs_modules=['scatter', 'bar', 'maps']` which would overwrite the `include_plotlyjs` argument? 
cc @etpinard

Comment (etpinard) [inline plotly/offline/offline.py]: :beers: 

maybe also a `include_mathjax` flag?

**Comment by theengineear** [comment]
First pass looks good to me too!!

**Comment by Immortalin** [comment]
@chriddyp merge this?

**Comment by chriddyp** [comment]
hey @cldougl @theengineear - I just added tests. Could you take another look? Would like to merge this early next week :)

**Comment by theengineear** [comment]
Sweet! Thanks for adding tests! :dancer: after you update the plot schema and tests :flags:

**Comment by cldougl** [comment]
same here- :dancer: after update/tests pass

**Comment by jackgolding** [comment]
Thanks heaps for this @chriddyp and all the testers!

**Comment by brovercleveland** [comment]
Excellent! Thanks @chriddyp!

**Pull request merged**